fuel/server.py

```diff
@@ -0,0 +1,177 @@
+import logging
+from collections import deque
+from multiprocessing import Process
+
+import numpy
+import six
+import zmq
+
+from fuel.utils import buffer_
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level='INFO')
+
+
+def send_arrays(socket, arrays, flags=0, copy=True, track=False, stop=False):
+    """Send a NumPy array using the buffer interface and some metadata.
+
+    Parameters
+    ----------
+    socket : :class:`zmq.Socket`
+        The socket to send data over.
+    arrays : list
+        A list of :class:`numpy.ndarray` to transfer.
+    stop : bool, optional
+        Instead of sending a series of NumPy arrays, send a JSON object
+        with a single `stop` key. The :func:`recv_arrays` will raise
+        ``StopIteration`` when it receives this.
+
+    See Also
+    --------
+    :meth:`zmq.Socket.send` for other arguments
+
+    Notes
+    -----
+    The protocol is very simple: A single JSON object that contains the
+    shape and data type for each array is transferred first. Subsequently
+    the arrays are sent as bytestreams (through NumPy's support of the
+    buffering protocol).
+
+    """
+    if stop:
+        mds = {'stop': True}
+        return socket.send_json(mds, flags)
+    else:
+        mds = [{'dtype': str(array.dtype), 'shape': array.shape}
```

Comment:
Thanks for the link to the NPY format. I ended up just using the `numpy.lib.format.header_data_from_array_1_0` function, which is the function the NPY format uses to extract the necessary metadata from an array. Added support for Fortran arrays in the process.

I'll leave the format specifier for a later PR if you think that's okay. I think it's clear enough from the function right now that it just supports NumPy arrays. Anything more exotic will probably just crash, like it should.